src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java

```diff
@@ -196,11 +197,26 @@ private List<LinkedFileViewModel> findAssociatedNotLinkedFiles(BibEntry entry) {
         return result;
     }
 
+    public void downloadFile(String urlText) {
+        try {
+            URL url = new URL(urlText);
+            addFromURL(url);
+        } catch (MalformedURLException exception) {
+            dialogService.showErrorDialogAndWait(
+                    Localization.lang("Invalid URL"),
+                    exception);
+        }
+    }
+
     public void fetchFulltext() {
         FulltextFetchers fetcher = new FulltextFetchers(
                 preferences.getImportFormatPreferences(),
                 preferences.getImporterPreferences());
-        BackgroundTask
+        Optional<String> urlField = entry.getField(StandardField.URL);
+        if (urlField.isPresent()) {
+            downloadFile(urlField.get());
+        } else {
```

Comment:
If `downloadFile` fails, JabRef should still fireup the BackgroundTask.

Maybe have `downloadFile` returning a `Boolean`? `true` if it went fine, `false` if it did not. 

here then if the result is `false`, start the Background task.